src/rs232.rs

```diff
@@ -103,9 +103,10 @@ impl RS232 {
     fn verify(& mut self) -> bool {
         let mut buffer = String::new();
 
-        for _ in 1..3 {
+        for _ in 1..5 {
             if let Ok(_) = self.read_line(& mut buffer) {
-                if buffer.starts_with("$GP") {
+                // check to see if data resembles NMEA messages
+                if buffer.len() >= 15 && buffer.starts_with("$G") && buffer.chars().nth(6).unwrap() == ',' {
```

Comment:
looks good. This line exceeds 100 chars limit though. Could you run rustfmt on it? Actually this reminded me of needing to run rustfmt on all source files, which I now did. You'd want to rebase all your changes now though but this one should be not a problem at least.